src/Symfony/Component/Translation/Bridge/Loco/LocoProvider.php

```diff
@@ -91,92 +84,137 @@ public function read(array $domains, array $locales): TranslatorBag
 
-        foreach ($translatorBag->getCatalogues() as $catalogue) {
-            foreach ($catalogue->all() as $messages) {
-                foreach ($messages as $id => $message) {
-                    if (\in_array($id, $deletedIds, true)) {
-                        continue;
-                    }
-
-                    $this->deleteAsset($id);
-                    $deletedIds[] = $id;
-                }
+        if (!$catalogue) {
+            $catalogue = $translatorBag->getCatalogues()[0];
+        }
+
+        $responses = [];
+
+        foreach (array_keys($catalogue->all()) as $domain) {
+            foreach ($this->getAssetsIds($domain) as $id) {
+                $responses[$id] = $this->client->request('DELETE', sprintf('assets/%s.json', $id));
+            }
+        }
+
+        foreach ($responses as $key => $response) {
+            if (200 !== $response->getStatusCode()) {
+                $this->logger->error(sprintf('Unable to delete translation key "%s" to Loco: "%s".', $key, $response->getContent(false)));
             }
         }
     }
 
-    private function createAsset(string $id): void
+    /**
+     * Returns array of internal Loco's unique ids.
+     */
+    private function getAssetsIds(string $domain): array
     {
-        $response = $this->client->request('POST', 'assets', [
-            'body' => [
-                'name' => $id,
-                'id' => $id,
-                'type' => 'text',
-                'default' => 'untranslated',
-            ],
-        ]);
+        $response = $this->client->request('GET', 'assets', ['query' => ['filter' => $domain]]);
+
+        if (200 !== $response->getStatusCode()) {
+            $this->logger->error(sprintf('Unable to get assets from Loco: "%s".', $response->getContent(false)));
+        }
 
-        if (409 === $response->getStatusCode()) {
-            $this->logger->info(sprintf('Translation key "%s" already exists in Loco.', $id), [
-                'id' => $id,
+        return array_map(function ($asset) {
+            return $asset['id'];
+        }, $response->toArray(false));
+    }
+
+    private function createAssets(array $keys): array
+    {
+        $responses = $createdIds = [];
+
+        foreach ($keys as $key) {
+            $responses[$key] = $this->client->request('POST', 'assets', [
+                'body' => [
+                    'text' => $key,
+                    'type' => 'text',
+                    'default' => 'untranslated',
+                ],
             ]);
-        } elseif (201 !== $response->getStatusCode()) {
-            $this->logger->error(sprintf('Unable to add new translation key "%s" to Loco: (status code: "%s") "%s".', $id, $response->getStatusCode(), $response->getContent(false)));
         }
+
+        foreach ($responses as $key => $response) {
+            if (409 === $response->getStatusCode()) {
+                $this->logger->info(sprintf('Translation key "%s" already exists in Loco.', $key), [
```

Comment:
Be careful about that. In Symfony, a translation key is only unique inside its domain, not globally.

Reply:
Yes, moreover Loco seems to not support this status code anymore (https://localise.biz/api/#!/assets/createAsset)